scheduler/rank.go

```diff
@@ -206,10 +206,35 @@ OUTER:
 			continue
 		}
 
-		// Index the existing network usage
+		// Index the existing network usage.
+		// This should never collide, sice it represents the current state of
+		// the node. If it does collide though, it means we found a bug! So
+		// collect as much information as possible and send it back to the
+		// server.
 		netIdx := structs.NewNetworkIndex()
-		netIdx.SetNode(option.Node)
-		netIdx.AddAllocs(proposed)
+		if collide, reason := netIdx.SetNode(option.Node); collide {
+			iter.sendEvent(&EvalEvent{
+				Reason:   reason,
+				NetIndex: netIdx.Copy(),
+				Node:     option.Node.Copy(),
```

Comment:
I don't think this is necessary since the Node should be pulled from memdb and be immutable.